frontend/scripts/react-components/tool/mapbox-map/mapbox-map.component.jsx

```diff
@@ -0,0 +1,181 @@
+/* eslint-disable react/no-danger */
+import React, { useState, useRef, useEffect } from 'react';
+import { PropTypes } from 'prop-types';
+import { LayerManager, Layer } from 'layer-manager/dist/components';
+import { PluginMapboxGl } from 'layer-manager';
+import ReactMapGL, { NavigationControl, FlyToInterpolator } from 'react-map-gl';
+import cx from 'classnames';
+import { TOOL_LAYOUT, BASEMAPS } from 'constants';
+import Basemaps from 'react-components/tool/basemaps';
+import Legend from 'react-components/tool/legend';
+import { easeCubic } from 'd3-ease';
+import activeLayers from './layers/unit-layers';
+import Warnings from './mapbox-map-warnings';
+import { getContextualLayersTemplates, getRasterLayerTemplate } from './layers/contextual-layers';
+import { getLayerOrder } from './layers/layer-config';
+import { getBaseLayer } from './layers/base-layer';
+import 'react-components/tool/mapbox-map/mapbox-map.scss';
+import Tooltip from 'legacy/info-tooltip.component';
+
+function MapBoxMap(props) {
+  const {
+    defaultMapView,
+    toolLayout,
+    basemapId,
+    selectedMapContextualLayersData,
+    selectedMapDimensionsWarnings
+  } = props;
+  const tooltip = new Tooltip('.js-node-tooltip');
+
+  const [mapAttribution, setMapAttribution] = useState(null);
+  const [viewport, setViewport] = useState({ ...defaultMapView });
+  const [loaded, setLoaded] = useState(false);
+  const [flying, setFlying] = useState(false);
+  const [tooltipData, setTooltip] = useState(null);
+  const [hoveredGeoId, setHoveredGeoId] = useState(null);
+  const mapRef = useRef();
+  const mapContainerRef = useRef();
+  const fullscreen = toolLayout === TOOL_LAYOUT.right;
+  const minimized = toolLayout === TOOL_LAYOUT.right;
+
+  const updateViewport = updatedViewport => {
+    setViewport({
+      ...viewport,
+      ...updatedViewport
+    });
+  };
+
+  useEffect(() => {
+    setViewport({
+      ...viewport,
+      ...defaultMapView
+    });
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [defaultMapView]);
+
+  useEffect(() => {
+    if (tooltipData) {
+      const { x, y, name, values } = tooltipData;
+      tooltip.show(x, y, name, values);
+    } else {
+      tooltip.hide();
+    }
+  }, [tooltip]);
+
+  const onLoad = () => {
+    setLoaded(true);
+    // if (!isEmpty(bounds) && !!bounds.bbox) {
+    //   fitBounds();
+    // }
+  };
+
+  useEffect(() => {
+    if (loaded) {
+      const attributionNode = document.querySelector('.mapboxgl-ctrl-attrib-inner');
+      if (attributionNode) {
+        setMapAttribution(attributionNode.innerHTML);
+      }
+    }
+  }, [loaded]);
+
+  const baseLayerInfo = BASEMAPS[basemapId];
+  const baseLayer = getBaseLayer(baseLayerInfo);
+
+  const layerOrder = getLayerOrder(baseLayerInfo.id);
+  const onHover = e => {
+    const { features, center } = e;
+    if (features?.length) {
+      const geoFeature = features.find(f => f.source === 'brazil_municipalities'); // Temporary
```

Comment:
maybe add a `// TODO:` here so we don't forget about this in the future!

Reply:
Thanks! we won't forget it as we will have to import the different unit layers and check the interaction